sup3r/preprocessing/data_handling.py

```diff
@@ -2186,6 +2358,66 @@ def feature_registry(cls):
             'topography': TopoH5}
         return registry
 
+    def run_daily_averages(self):
+        """Calculate daily average data and store as attribute.
+
+        Note that the H5 clearsky ratio feature requires special logic to match
+        the climate change dataset of daily average GHI / daily average CS_GHI.
+        This target climate change dataset is not equivelant to the average of
+        instantaneous hourly clearsky ratios
+        """
+
+        msg = ('Data needs to be hourly with at least 24 hours, but data '
+               'shape is {}.'.format(self.data.shape))
+        assert self.data.shape[2] % 24 == 0, msg
+        assert self.data.shape[2] > 24, msg
+
+        n_data_days = int(self.data.shape[2] / 24)
+        daily_data_shape = (self.data.shape[0:2] + (n_data_days,)
+                            + (self.data.shape[3],))
+
+        logger.info('Calculating daily average datasets for {} training '
+                    'data days.'.format(n_data_days))
+
+        self.daily_data = np.zeros(daily_data_shape, dtype=np.float32)
+
+        self.daily_data_slices = np.array_split(np.arange(self.data.shape[2]),
+                                                n_data_days)
+        self.daily_data_slices = [slice(x[0], x[-1] + 1)
+                                  for x in self.daily_data_slices]
+
+        i_ghi = self.features.index('ghi')
+        i_cs = self.features.index('clearsky_ghi')
+        i_ratio = self.features.index('clearsky_ratio')
+
+        for d, t_slice in enumerate(self.daily_data_slices):
+            self.daily_data[:, :, d, i_ghi] = daily_temporal_coarsening(
+                self.data[:, :, t_slice, i_ghi], temporal_axis=2)[:, :, 0]
+            self.daily_data[:, :, d, i_cs] = daily_temporal_coarsening(
+                self.data[:, :, t_slice, i_cs], temporal_axis=2)[:, :, 0]
+
+            # note that this ratio of daily irradiance sums is not the same as
+            # the average of hourly ratios.
+            total_ghi = np.nansum(self.data[:, :, t_slice, i_ghi], axis=2)
+            total_cs_ghi = np.nansum(self.data[:, :, t_slice, i_cs], axis=2)
+            avg_cs_ratio = total_ghi / total_cs_ghi
+            self.daily_data[:, :, d, i_ratio] = avg_cs_ratio
+
+        # remove ghi and clearsky ghi from feature set. These shouldn't be used
+        # downstream for solar cc and keeping them confuses the batch handler
+        logger.info('Finished calculating daily average clearsky_ratio, '
+                    'removing ghi and clearsky_ghi from the '
+                    'DataHandlerH5SolarCC feature list.')
+        ifeats = np.array([i for i in range(len(self.features))
```

Comment:
maybe we could add something like COMPUTE_ONLY_FEATURES = (...) and make this removal routine more general. could be placed in the super class also.

Reply:
So these features are extra weird because they're really required to compute the daily average data, not just for computing the high-res data. So i could generalize this to the top level CC class, but it really does seem like a one-off at this point.

Reply:
Ah yeah, ok makes sense.